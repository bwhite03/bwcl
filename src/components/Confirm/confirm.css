.confirm-delete-overlay {
  position: fixed;
  top: 0;
  left: 0;
  z-index: 1040;
  width: 100vw;
  height: 100vh;
  background-color: #000;
  opacity: 0.5;
}

.confirm-delete-wrapper {
  position: fixed;
  top: 0;
  left: 0;
  z-index: 1050;
  width: 100%;
  height: 100%;
  overflow-x: hidden;
  overflow-y: hidden;
  outline: 0;
}

.confirm-delete {
  z-index: 100;
  background: #fff;
  position: relative;
  margin: 1.75rem auto;
  border-radius: 10px;
  max-width: 500px;
}

.confirm-delete-header {
  display: 'flex';
  border-bottom: 1px solid #000;
  background-color: #1f1e2e;
  color: #fff;
  border-top-left-radius: 10px;
  border-top-right-radius: 10px;
  padding: 5px;
}

.confirm-delete-title {
  font-size: 1.3rem;
  font-weight: bold;
  text-align: left;
  margin-left: 10px;
}

.confirm-delete-close-box {
  font-size: 1.4rem;
  font-weight: 700;
  line-height: 1;
  position: absolute;
  right: 10px;
}

.confirm-delete-close-button {
  color: #000;
  opacity: 0.3;
  cursor: pointer;
  padding-bottom: 3px;
}

.confirm-delete-body {
  padding: 10px;
  background-color: #27293d;
  color: #fff;
  text-align: center;
}

.confirm-delete-buttons {
  display: grid;
  grid-template-columns: 1fr 1fr;
  text-align: center;
  margin-top: 10px;
}

.confirm-delete-body input {
  padding: 4px 8px;
  border-radius: 5px;
  margin-left: 8px;
}

.confirm-delete-body input:focus {
  outline: none;
}

.confirm-delete-error {
  margin-top: 8px;
  color: red;
  text-align: center;
  font-size: 1.2rem;
}

.confirm-delete-cancel-button {
  background-color: red;
  color: #fff;
  outline: 0;
  border: 0;
  margin-left: 10px;
}

.confirm-delete-confirm-button {
  background-color: limegreen;
  color: #fff;
  outline: 0;
  border: 0;
}

.more-info {
  color: orange;
  margin-bottom: 8px;
}
