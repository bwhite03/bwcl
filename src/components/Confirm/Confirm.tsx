import React, { useState } from 'react';
import { createPortal } from 'react-dom';
import './confirm.css';

export interface ConfirmProps {
  isShowing: boolean;
  hide: () => void;
  title: string;
  moreInfo?: string;
  confirmText: string;
  setSuccess: (b: boolean) => void;
}

const Confirm = (props: ConfirmProps) => {
  const [input, setInput] = useState('');
  const [error, setError] = useState('');
  const {
    isShowing,
    hide,
    title = 'Confirm Delete',
    moreInfo,
    confirmText,
    setSuccess,
  } = props;

  const compareDeleteText = () => {
    if (input === confirmText) {
      setSuccess(true);
      hide();
    } else {
      setError(`${input} does not match ${confirmText}`);
    }
  };

  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    setInput(e.target.value);
  };

  return (
    <React.Fragment>
      {isShowing
        ? createPortal(
            <React.Fragment>
              <div className="confirm-delete-overlay" />
              <div
                className="confirm-delete-wrapper"
                aria-modal
                aria-hidden
                tabIndex={-1}
                role="dialog"
              >
                <div className="confirm-delete">
                  <div className="confirm-delete-header">
                    <span className="confirm-delete-title">{title}</span>
                    <span className="confirm-delete-close-box">
                      <button
                        type="button"
                        className="confirm-delete-close-button"
                        data-dismiss="modal"
                        aria-label="close"
                        onClick={hide}
                      >
                        <span aria-hidden="true">&times;</span>
                      </button>
                    </span>
                  </div>
                  <div className="confirm-delete-body">
                    <p>
                      Type{' '}
                      <i>
                        <b>{confirmText}</b>
                      </i>{' '}
                      in the box to confirm.
                    </p>
                    {moreInfo && moreInfo.length > 0 ? (
                      <div className="more-info">{moreInfo}</div>
                    ) : null}
                    <div>
                      <div className="form-group">
                        <input
                          type="text"
                          data-testid="confirm-input"
                          className="confirm-delete-input"
                          id="txtDeleteConfirm"
                          value={input}
                          onChange={handleInputChange}
                          autoComplete="off"
                        />
                        {error.length > 0 && (
                          <div className="confirm-delete-error">{error}</div>
                        )}
                        <div className="confirm-delete-buttons">
                          <button
                            className="confirm-delete-cancel-button"
                            onClick={hide}
                          >
                            Cancel
                          </button>
                          <button
                            className="confirm-delete-confirm-button"
                            onClick={compareDeleteText}
                          >
                            Confirm
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </React.Fragment>,
            document.body
          )
        : null}
    </React.Fragment>
  );
};

export default Confirm;
