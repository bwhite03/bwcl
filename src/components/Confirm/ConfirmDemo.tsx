import React, { useState, useEffect } from 'react';
import Confirm from './Confirm';
import './confirm.css';

const ConfirmDemo = () => {
  const [isShowing, setIsShowing] = useState(false);
  const [success, setSuccess] = useState(false);

  useEffect(() => {
    if (success) {
      handleDelete();
    }
  });

  const hide = () => {
    setIsShowing(false);
  };

  const handleDelete = () => {
    console.log('delete something');
  };

  return (
    <div>
      <div>
        <button onClick={() => setIsShowing(!isShowing)}>
          Show Confirm Delete
        </button>
      </div>
      <Confirm
        title="Delete"
        isShowing={isShowing}
        hide={hide}
        confirmText={'delete me'}
        // @ts-ignore
        setSuccess={success}
      />
    </div>
  );
};

export default ConfirmDemo;
