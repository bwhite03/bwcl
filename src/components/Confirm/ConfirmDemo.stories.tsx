import React, { Component } from 'react';
import { Story, Meta } from '@storybook/react';
import ConfirmDemo from './ConfirmDemo';

export default {
  title: 'Confirm',
  component: ConfirmDemo,
} as Meta;

const Template: Story = (args) => <ConfirmDemo {...args} />;

export const ConfirmDemoExample = Template.bind({});

ConfirmDemoExample.args = {};
