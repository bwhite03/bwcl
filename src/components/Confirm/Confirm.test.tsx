import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import Confirm from './Confirm';

describe('Confirm', () => {
  test('should render correctly', () => {
    const hide = jest.fn();
    const setSuccess = jest.fn();
    const { container } = render(
      <Confirm
        title={'Delete Me'}
        isShowing={true}
        hide={hide}
        confirmText="delete me"
        setSuccess={setSuccess}
      />
    );
    expect(container).toMatchSnapshot();
  });
  test('should render without a title', () => {
    const hide = jest.fn();
    const setSuccess = jest.fn();
    const { container } = render(
      // @ts-ignore
      <Confirm
        isShowing={true}
        hide={hide}
        confirmText="delete me"
        setSuccess={setSuccess}
      />
    );
    expect(container).toMatchSnapshot();
  });
  test('should render with more info', () => {
    const hide = jest.fn();
    const setSuccess = jest.fn();
    const { container } = render(
      // @ts-ignore
      <Confirm
        isShowing={true}
        hide={hide}
        confirmText="delete me"
        setSuccess={setSuccess}
        moreInfo={'This operation cannot be undone'}
      />
    );
    expect(container).toMatchSnapshot();
  });
  test('should handle input and errors', () => {
    const hide = jest.fn();
    const setSuccess = jest.fn();
    const { container, getByTestId, getByText } = render(
      <Confirm
        title="delete something"
        isShowing={true}
        hide={hide}
        confirmText="delete me"
        setSuccess={setSuccess}
        moreInfo={'This operation cannot be undone'}
      />
    );
    userEvent.type(screen.getByTestId('confirm-input'), 'dele');
    const btn = screen.getByText('Confirm');
    fireEvent.click(btn);
    const error = screen.getByText('dele does not match delete me');
    expect(error).toBeInTheDocument();
  });
  test('should confirm without errors', () => {
    const hide = jest.fn();
    const setSuccess = jest.fn();
    const { container, getByTestId, getByText } = render(
      <Confirm
        title="delete something"
        isShowing={true}
        hide={hide}
        confirmText="delete me"
        setSuccess={setSuccess}
        moreInfo={'This operation cannot be undone'}
      />
    );
    userEvent.type(screen.getByTestId('confirm-input'), 'delete me');
    const btn = screen.getByText('Confirm');
    fireEvent.click(btn);
  });
  test('should not show', () => {
    const hide = jest.fn();
    const setSuccess = jest.fn();
    const { container, getByTestId, getByText } = render(
      <Confirm
        title="delete something"
        isShowing={false}
        hide={hide}
        confirmText="delete me"
        setSuccess={setSuccess}
        moreInfo={'This operation cannot be undone'}
      />
    );
  });
});
