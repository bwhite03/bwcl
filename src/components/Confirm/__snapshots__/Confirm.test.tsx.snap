// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirm not pass a title 1`] = `<div />`;

exports[`Confirm render with more info 1`] = `<div />`;

exports[`Confirm should render correctly 1`] = `<div />`;

exports[`Confirm should render with more info 1`] = `<div />`;

exports[`Confirm should render without a title 1`] = `<div />`;
